module/zfs/abd.c

```diff
@@ -215,15 +221,37 @@ abd_fini(void)
 		kstat_delete(abd_ksp);
 		abd_ksp = NULL;
 	}
+}
+
+#else
 
-	kmem_cache_destroy(abd_chunk_cache);
-	abd_chunk_cache = NULL;
+struct page;
+#define	kpm_enable			1
+#define	abd_alloc_chunk() \
+	((struct page *)kmem_alloc(PAGESIZE, KM_SLEEP))
+#define	abd_free_chunk(chunk) 		kmem_free(chunk, PAGESIZE)
```

Comment:
Prior to this change PAGESIZE wasn't used extensively in userspace so it was implemented in `lib/libspl/include/sys/param.h` just as `sysconf(_SC_PAGESIZE)`.  That's not going to work well if use PAGESIZE everywhere.  We're either going to want to implement PAGESIZE differently or alternately preserve the  `zfs_abd_chunk_size` and initialize it once in `abd_init()`.

Potentially we could also preserve the kmem cache chunk case for use solely by user space.  That would help minimize our delta with upstream.